### PR TITLE
powerdns-admin: fix build

### PR DESCRIPTION
**Pull request by Flakebi**
## Description of changes

Fix the build of flask-seasurf and powerdns-admin.
Things got broken by the last flask and werkzeug update.

`powerdns-admin.tests` succeed with these fixes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
@ofborg eval

File: pkgs/applications/networking/powerdns-admin/default.nix [hunk 1]
```diff
@@ -37,6 +42,7 @@ let
     nativeBuildInputs = pythonDeps;
     patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/powerdns-admin/default.nix:43]:
```suggestion
    postPatch = ''
```

File: pkgs/applications/networking/powerdns-admin/default.nix [hunk 2]
```diff
@@ -95,6 +101,16 @@ in stdenv.mkDerivation {
...
+
+    # flask 2.3: 'Blueprint' object has no attribute 'before_app_first_request'
+    # Run code before app.run() instead
+    substituteInPlace run.py \
+      --replace 'from powerdnsadmin import create_app' 'from powerdnsadmin import create_app
+    from .powerdnsadmin.routes.index import register_modules' \
+      --replace 'app = create_app()' 'app = create_app()
+        with app.app_context():
+            register_modules()'
+    ${all_post_patch}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/powerdns-admin/default.nix:113]: We are accumulating a rather large amount of substituteInPlace here which change code. We should really convert them to patches to notice breakages and changes in the code immediately.

File: pkgs/development/python-modules/flask-seasurf/default.nix [hunk 3]
```diff
@@ -12,6 +12,29 @@ buildPythonPackage rec {
...
+  postPatch = ''
+    # - cookie_jar is private in werkzeug 2.3, so recreate the client instead
+    # - set_cookie does not take a hostname argument anymore, use domain instead
+    substituteInPlace test_seasurf.py \
+      --replace '    client.cookie_jar.clear()' 'with self.app.test_client() as client:' \
+      --replace "client.set_cookie('www.example.com', self.csrf._csrf_name, token)" "client.set_cookie(self.csrf._csrf_name, token, domain='www.example.com')" \
+      --replace "c.delete('/foo/baz'," "c.delete('/foo/baz', content_type='application/json'," \
+
+    # Headers need to specify a content type
+    sed -i "s#\(post\|delete\)(\(u\?'[^']*'\(, headers=headers\)\?\))#\1(\2, content_type='application/json')#g" test_seasurf.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-seasurf/default.nix:32]: Same here, this sed is rather complex and could easily break in the future

File: pkgs/development/python-modules/flask-seasurf/default.nix [hunk 4]
```diff
@@ -12,6 +12,29 @@ buildPythonPackage rec {
...
+
+    # Disable some tests
+    sed -i "s#\(\(test_header_set_on_post\|test_https_good_referer\|test_https_referer_check_disabled\)(self):\)#\1\n        return#g" test_seasurf.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-seasurf/default.nix:35]: Do they really not support pytest?

**Comment by pbsds** [comment]
ZHF #264232, fixes:
* https://hydra.nixos.org/build/238947444
* https://hydra.nixos.org/build/240846908

File: pkgs/development/python-modules/flask-seasurf/default.nix (hunk 4 above)
Comment (pbsds) [reply pkgs/development/python-modules/flask-seasurf/default.nix:35]: The fix is commonly to add the disabled tests to `disabledTests`

Comment (Flakebi) [reply pkgs/development/python-modules/flask-seasurf/default.nix:35]: Sorry, didn’t have time in the last weeks to look at this again (not yet sure if I will find time in the coming weeks).

I tried `disabledTests` and it didn’t work (I don’t remember which test system this uses but I didn’t even find a command line flag to skip tests).

**Comment by pbsds** [comment]
@ofborg eval

**Pull request merged**